base/compiler/tfuncs.jl

```diff
@@ -1759,9 +1770,16 @@ const _tvarnames = Symbol[:_A, :_B, :_C, :_D, :_E, :_F, :_G, :_H, :_I, :_J, :_K,
             uncertain = true
             unw = unwrap_unionall(ai)
             isT = isType(unw)
-            if isT && isa(ai,UnionAll) && contains_is(outervars, ai.var)
-                ai = rename_unionall(ai)
-                unw = unwrap_unionall(ai)
+            if isT
+                tai = ai
+                while isa(tai, UnionAll)
+                    if contains_is(outervars, tai.var)
+                        ai = rename_unionall(ai)
+                        unw = unwrap_unionall(ai)
+                        break
```

Comment:
```suggestion
```
?

Reply:
IIUC, `rename_unionall` replace all typevars in `ai` thus there's no need to re-renaming?

Reply:
Ah, okay. That function is a bit awkward then, since it also exists in C without that behavior, and this is the only usage of the function anywhere (including not existing in any tests)